File: LLM-Translations/GPT-4/codenet/Python/Go/s760938927.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var h, w, ka int
	fmt.Scan(&h, &w, &ka)

	mat := make([]string, h)
	for i := 0; i < h; i++ {
		fmt.Scan(&mat[i])
	}

	ans := 0
	for i := 0; i < (1 << (h + w)); i++ {
		b := strconv.FormatInt(int64(i), 2)
		b = fmt.Sprintf("%0*s", h+w, b)
		rs := b[:h]
		cs := b[h:]
		ct := 0
		for l := 0; l < h; l++ {
			for k := 0; k < w; k++ {
				if mat[l][k] == '#' && rs[l] == '0' && cs[k] == '0' {
					ct++
				}
			}
		}
		if ct == ka {
			ans++
		}
	}
	fmt.Println(ans)
}

// 

File: LLM-Translations/GPT-4/codenet/Python/C/s760938927.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    int h, w, ka;
    scanf("%d %d %d", &h, &w, &ka);

    char mat[h][w];
    for (int i = 0; i < h; i++) {
        scanf("%s", mat[i]);
    }

    int ans = 0;
    for (int i = 0; i < (1 << (h + w)); i++) {
        char b[h + w + 1];
        sprintf(b, "%0*d", h + w, atoi(_itoa(i, b, 2)));
        char *rs = b;
        char *cs = b + h;
        int ct = 0;

        for (int l = 0; l < h; l++) {
            for (int k = 0; k < w; k++) {
                if (mat[l][k] == '#' && rs[l] == '0' && cs[k] == '0') {
                    ct++;
                }
            }
        }

        if (ct == ka) ans++;
    }

    printf("%d\n", ans);

    return 0;
}
// 
